LanguageFeatures/Static-access-shorthand/constant_expression_A07_t01.dart

```diff
@@ -31,20 +31,20 @@ extension type const ET<X, T>(int _) {
 }
 
 void foo<T>() {
-  C<int, T> c1 = const .new();
-//                     ^^^^
+  C<int, void Function<Y extends T>()> c1 = const .new();
```

Comment:
This will stop working if we decide that `const .new()` should be inferred as `const C<int, Never>.new()`, which would not have any errors.

I guess we can just proceed for now (keeping this test as it is with this PR), and handle the missing compile-time error later if constant-expression inference is changed in this manner.

Reply:
Similarly for the other cases.